### PR TITLE
[SMALLFIX] Add lineage integration test

### PR DESCRIPTION
**Pull request by aaudiber**
Lineage context needs to be reset whenever client context is reset because lineage context depends on the value of `ClientContext.getMasterAddress()`. Without this change, all tests use the cluster created for the first lineage test.

### CODE REVIEW

**Comment by aaudiber** [comment]
Jenkins, test this please.

File: integration-tests/src/test/java/tachyon/master/lineage/LineageMasterIntegrationTest.java [hunk 1]
```diff
@@ -152,6 +164,39 @@ public void lineageCompleteAndAsyncPersistTest() throws Exception {
     }
   }
 
+  /**
+   * Tests that a lineage job is executed when the output file for the lineage is reported as lost.
+   */
+  @Test(timeout = 20000)
+  public void lineageRecoveryTest() throws Exception {
+    File logFile = mFolder.newFile();
+    logFile.delete();
```
Comment (yupeng9) [inline integration-tests/src/test/java/tachyon/master/lineage/LineageMasterIntegrationTest.java]: why is this needed?

File: clients/unshaded/src/main/java/tachyon/client/lineage/LineageContext.java [hunk 2]
```diff
@@ -53,4 +53,15 @@ public LineageMasterClient acquireMasterClient() {
   public void releaseMasterClient(LineageMasterClient masterClient) {
     mLineageMasterClientPool.release(masterClient);
   }
+
+  /**
+   * Re-initializes the {@link LineageContext}. This method should only be used in
+   * {@link ClientContext}.
+   */
+  public void reset() {
+    if (mLineageMasterClientPool != null) {
```
Comment (yupeng9) [inline clients/unshaded/src/main/java/tachyon/client/lineage/LineageContext.java]: Do we need to synchronize on `mLineageMasterClientPool`

File: integration-tests/src/test/java/tachyon/master/lineage/LineageMasterIntegrationTest.java (hunk 1 above)
Comment (aaudiber) [inline integration-tests/src/test/java/tachyon/master/lineage/LineageMasterIntegrationTest.java]: we need it so that we can wait on `logFile.exists()` below. I've added comments to make this less confusing.

**Comment by calvinjia** [comment]
LGTM

File: clients/unshaded/src/main/java/tachyon/client/lineage/LineageContext.java (hunk 2 above)
Comment (aaudiber) [inline clients/unshaded/src/main/java/tachyon/client/lineage/LineageContext.java]: I think it's fine for it to be unsynchronized since reset is a temporary hack to allow tests to reset the configuration. This method should never be called while there are multiple threads trying to access the `LineageContext`. I've created https://tachyon.atlassian.net/browse/TACHYON-1623 to address the issue.

**Comment by yupeng9** [comment]
LGTM. Thanks for adding the test coverage.
